### PR TITLE
Bug Fixes

### PR DESCRIPTION
**Pull request by tjklemz**
Fixed major bugs which caused the app to crash.
Fixed Mac/Unix related bugs due to file separators.
Fixed new lines (made them uniform—everything is LF instead of CRLF which is outdated).
Fixed live reload (in DEBUG_MODE=1).
Fixed window autofocus issue in live reload.

Refactored a few minor things (mostly due to above changes).

Made `gulp` run the `lint` and `test` tasks instead of `build`. (No need to run `gulp lint && gulp test`—can just run `gulp`.)

### CODE REVIEW

File: app/js/bootstrap.js [hunk 1]
```diff
@@ -33,23 +33,28 @@
...
             let win = this.window;
-            win.show();
-            // NOTE: needs to be in a setTimeout, otherwise doesn't work properly
-            setTimeout(win.focus.bind(win), 1);
+
+            if (!win.shown) {
+                win.show();
+                win.shown = true;
+                // NOTE: needs to be in a setTimeout, otherwise doesn't work properly
+                setTimeout(win.focus.bind(win), 1);
```
Comment (neutrinog) [inline app/js/bootstrap.js]: hm.. this is kinda hacky don't they have a callback or something?

Comment (tjklemz) [inline app/js/bootstrap.js]: What's hacky?

Comment (neutrinog) [inline app/js/bootstrap.js]: the timeout.

Comment (tjklemz) [inline app/js/bootstrap.js]: https://www.youtube.com/watch?v=8aGhZQkoFbQ

**Pull request merged**